### PR TITLE
fix(gateway-client): remove unused methods and `MaybePendingBlock`

### LINKED ISSUE #1784: Remove `MaybePendingBlock` type
Its `serde::Deserialize` implementation is broken anyway and the recent gateway API changes to split pending block related calls from the rest enable us to just remove this abstraction.

### PR DESCRIPTION
**Pull request by kkovaacs**
Deserialization for `MaybePendingBlock` was broken anyway, and since the recent changes to use a separate method for fetching the pending state update and block we no longer need this type.

This PR also removes the `_deprecated` methods from the gateway client API -- remaining callers in tests and `chain()` were updated.

Closes #1784

### CODE REVIEW

**Review by CHr15F0x — APPROVED** [review]

File: crates/gateway-client/src/lib.rs
Comment (Mirko-von-Leipzig) [inline crates/gateway-client/src/lib.rs:1]: Should the state update and block tests not combine to create tests for the `pending_block` and `state_update_with_block` methods?

I realise they weren't actually tested before, but that's really an oversight and not by design.

File: crates/gateway-client/src/lib.rs [hunk 1]
```diff
@@ -61,10 +47,6 @@ pub trait GatewayApi: Sync {
         unimplemented!();
     }
 
-    async fn state_update_deprecated(&self, block: BlockId) -> Result<StateUpdate, SequencerError> {
-        unimplemented!();
-    }
-
     async fn state_update_with_block(
```
Comment (Mirko-von-Leipzig) [inline crates/gateway-client/src/lib.rs:50]: Now that the names are free..

```suggestion
    async fn block(
```

**Review by Mirko-von-Leipzig — APPROVED** [review]

File: crates/gateway-client/src/lib.rs
Comment (kkovaacs) [reply crates/gateway-client/src/lib.rs:1]: Yep, that's a good opportunity. Some limitations apply there though, as `state_update_with_block` requires a block number as input.

Comment (Mirko-von-Leipzig) [reply crates/gateway-client/src/lib.rs:1]: Yeah only really need one maybe :)

File: crates/gateway-client/src/lib.rs (hunk 1 above)
Comment (kkovaacs) [reply crates/gateway-client/src/lib.rs:50]: Well, I'm not quite sure. `block` is not a great description of what this method does?

Comment (Mirko-von-Leipzig) [reply crates/gateway-client/src/lib.rs:48]: It gets the block's data :D? Ofcourse a bit confusing given the history, but if we were just starting fresh now I would have just called this method get `block` 🤷 No big deal though.

Maybe `full_block` then?

Comment (kkovaacs) [reply crates/gateway-client/src/lib.rs:48]: It doesn't even get _all_ data for the block. For example, signature's not included...

**Pull request merged**